feature_engine/discretisation/geometric_width.py

```diff
@@ -0,0 +1,161 @@
+    _transform_discretiser_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine._variable_handling.init_parameter_checks import (
+    _check_init_parameter_variables,
+)
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+
+
+@Substitution(
+    return_object=_return_object_docstring,
+    return_boundaries=_return_boundaries_docstring,
+    precision=_precision_docstring,
+    binner_dict_=_binner_dict_docstring,
+    fit=_fit_discretiser_docstring,
+    transform=_transform_discretiser_docstring,
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class GeometricWidthDiscretiser(BaseDiscretiser):
+    """
+    The GeometricWidthDiscretiser() divides continuous numerical variables into
+    intervals of increasing width with equal increments. Note that the
+    proportion of observations per interval may vary.
+
+    Sizes (a) of n intervals will follow geometric progression
+
+    .. math::
+        a_i+1 = a_i r^(n+i)
+
+    The GeometricWidthDiscretiser() works only with numerical variables.
+    A list of variables can be passed as argument. Alternatively, the discretiser
+    will automatically select all numerical variables.
+
+    The GeometricWidthDiscretiser() first finds the boundaries for the intervals for
+    each variable. Then, it transforms the variables, that is, sorts the values into
+    the intervals.
+
+    Note: this discretiser needs higher precision, because bins width could be very
+    small. With smaller range of the binned values and/or higher number of bins,
+    smaller precision values could be needed.
+
+    More details in the :ref:`User Guide <increasing_width_discretiser>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    bins: int, default=10
+        Desired number of intervals / bins.
+
+    {return_object}
+
+    {return_boundaries}
+
+    {precision}
+
+    Attributes
+    ----------
+    {binner_dict_}
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {transform}
+
+    References
+    ----------
+    .. [1]  J. Reiser, "Classification Systems",
+        https://www.slideshare.net/johnjreiser/classification-systems
+
+    .. [2] Geometric Interval Classification
+        http://wiki.gis.com/wiki/index.php/Geometric_Interval_Classification
+
+    .. [3] Geometric progression
+        https://en.wikipedia.org/wiki/Geometric_progression
+
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        bins: int = 10,
+        return_object: bool = False,
+        return_boundaries: bool = False,
+        precision: int = 7,
+    ):
+
+        if not isinstance(bins, int):
```

Comment:
here we probably also need a check that bins > 0? or we leave that to pandas.cut?

Reply:
We could leave all checks of precision to pandas, no need to duplicate it